android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -1140,6 +1144,29 @@ constructor(
     return null
   }
 
+  suspend fun retrieveFlattenedSubLocations(locationId: String): LinkedList<Location> {
+    val locations = LinkedList<Location>()
+    val resources: LinkedList<Location> = retrieveSubLocations(locationId)
+
+    while (resources.isNotEmpty()) {
+      val currentResource = resources.removeFirst()
+      locations.add(currentResource)
+      retrieveSubLocations(currentResource.logicalId).forEach(resources::addLast)
+    }
+    return locations
+  }
+
+  private suspend fun retrieveSubLocations(locationId: String) =
+    withContext(dispatcherProvider.io()) {
+      fhirEngine
+        .search<Location>(
+          Search(type = ResourceType.Location).apply {
+            filter(Location.PARTOF, { value = "Location/$locationId" })
```

Comment:
Not sure what I feel about the hardcode

Reply:
For this specific implementation, we are retrieving the sublocations. Is any other property used to link 2 locations other than `partOf` attribute? Or are you referring to the reference part?  I can update it to the code snippet below to achieve the same thing:

`locationId.asReference(ResourceType.Location)`